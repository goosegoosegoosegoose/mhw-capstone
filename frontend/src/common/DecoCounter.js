import React, { useState, useContext, useEffect } from "react";
import UserContext from "../auth/userContext";

const DecoCounter = ({id, c}) => {
  const currentUser = useContext(UserContext);
  const [counter, setCounter] = useState((currentUser ? (currentUser.decorations ? (currentUser.decorations[id] ? currentUser.decorations[id] : 0) : 0) : 0));
  let change = (currentUser ? currentUser.decorations : null)

  useEffect(() => {
    const updateCount = () => {
      if (currentUser) {
        if (currentUser.decorations) {
          if (!currentUser.decorations[id]) {
            setCounter(0)
          } else {
            setCounter(currentUser.decorations[id])
          }
        }
      }
    }
    updateCount();
  }, [change]);

  return (
    <>
      <span className={c} >{counter}</span>
    </>
  )
}

export default DecoCounter;