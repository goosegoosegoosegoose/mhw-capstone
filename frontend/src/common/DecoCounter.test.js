import { render } from '@testing-library/react';
import DecoCounter from "./DecoCounter";

test('smoke test', () => {
  render(      
    <DecoCounter />
  );
});

test('snapshot', () => {
  const { asFragment } = render(<DecoCounter />);
  expect(asFragment()).toMatchSnapshot();
});