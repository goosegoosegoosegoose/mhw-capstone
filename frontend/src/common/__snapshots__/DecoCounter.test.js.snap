// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <span>
    0
  </span>
</DocumentFragment>
`;
